tensornetwork/contractors/custom_path_solvers/nconinterface.py

```diff
@@ -0,0 +1,227 @@
+  Returns:
+    np.ndarray: the log-adjacency matrix.
+  """
+  # process inputs  
+  N = len(connects)
+  ranks = [len(connects[i]) for i in range(N)]
+  flat_connects = np.hstack([connects[i] for i in range(N)])
+  tensor_counter = np.hstack([i*np.ones(ranks[i],dtype=int) for i in range(N)])
+  index_counter = np.hstack([np.arange(ranks[i]) for i in range(N)])
+  
+  # build log-adjacency index-by-index  
+  log_adj = np.zeros([N,N]);
+  unique_connects = np.unique(flat_connects)
+  for i in range(len(unique_connects)):
+      # identify tensor/index location of each edge
+      tnr = tensor_counter[flat_connects == unique_connects[i]]
+      ind = index_counter[flat_connects == unique_connects[i]]
+      if len(ind) == 1: # external index
+        log_adj[tnr[0],tnr[0]] += np.log10(tensors[tnr[0]].shape[ind[0]])
+      elif len(ind) == 2: # internal index
+        if (tnr[0] != tnr[1]): # ignore partial traces
+          log_adj[tnr[0],tnr[1]] += np.log10(tensors[tnr[0]].shape[ind[0]])
+          log_adj[tnr[1],tnr[0]] += np.log10(tensors[tnr[0]].shape[ind[0]])
+   
+  return log_adj
+
+#------------------------------------------------------------------------#
+def ord_to_ncon(connects: List[List[int]],
+                orders: np.ndarray):
+  """
+  Produces a `ncon` compatible index contraction order from the sequence of 
+  pairwise contractions. 
+  Args:
+    connects: list of the tensor connections (in standard `ncon` format).
+    orders: array of dim (2,N-1) specifying the set of N-1 pairwise 
+      tensor contractions.
+  Returns:
+    np.ndarray: the contraction order (in `ncon` format).
+  """
+  
+  N = 1 + (orders.size//2)
+  orders = orders.reshape(2,N-1)
+  new_connects = [connects[i] for i in range(N)]
+  cont_order = np.zeros([0],dtype=int)
+  
+  for i in range(N): # always do partial traces first
+    uni_inds, counts = np.unique(new_connects[i],return_counts=True)
+    tr_inds = uni_inds[np.flatnonzero(counts == 2)]
+    cont_order = np.concatenate((cont_order,tr_inds))
+    temp_connect = np.array(new_connects[i])
+    for ind in tr_inds:
+      temp_connect = temp_connect[temp_connect != ind]
+    new_connects[i] = temp_connect 
+  
+  for i in range(N-1):  
+    # find common indices between tensor pair
+    cont_many, A_cont, B_cont = np.intersect1d(new_connects[orders[0,i]],
+                                               new_connects[orders[1,i]],  
+                                               return_indices=True)
+    temp_connects = np.append(np.delete(new_connects[orders[0,i]], A_cont), 
+                              np.delete(new_connects[orders[1,i]], B_cont))
+    cont_order = list(np.concatenate((cont_order,cont_many),axis=0))
+    
+    # build new set of connects
+    new_connects[orders[0,i]] = temp_connects
+    del new_connects[orders[1,i]]
+    
+  return cont_order
+
+#------------------------------------------------------------------------#
+def ncon_cost_check(tensor_list_in: List[np.ndarray], 
```

Comment:
can we call this just `tensors`?

Reply:
sure, I agree it should be that. I had just forgotten to change the name in this function.